### ISSUE TITLE
An invalid PHPExcel file causes catastrophic backtraces

### ISSUE THREAD

**Issue by PatrickRose** [PHPOffice/PHPExcel#1012]
PHPExcel's method of loading a file caused us to have a massive log file (70 gig!) due to the method it uses to load a file. The stack trace print points to `simplexml_load_string` being used in `PHPExcel_Reader_Excel2007::load()`, but I haven't been able to fully verify that (due to aforementioned 70 gigs of data). The file seemed to have multiple mismatching closing tags so each time `simplexml_load_string` found one we had a log entry which included the full backtrace and the argument list multiple times.

Is it possible to stream the contents or at least a temporary file and load it that way?

**Comment by johnkary** [comment]
@PatrickRose The large log file may be due to your PHP configuration's logging verbosity. The PHP configuration is often in a `php.ini` file whose location varies according to SAPI. Check PHP's [The configuration file](http://php.net/manual/en/configuration.file.php) documentation for more info.

Look for the directive [`error_reporting`](http://php.net/manual/en/function.error-reporting.php) for more information. This isn't something controlled specifically by PHPExcel but something you control in your environment.

Where PHPExcel might be able to change is instead using code that does not emit warning/error messages. Could you supply some of the warning messages?

**Comment by Quix0r** [comment]
No, reducing logging means hiding bugs in application. I also see tons of E_NOTICE for missing array keys (-1, -x, y) and then ending up in a out-of-memory condition (memory leak).

**Comment by Quix0r** [comment]
Here are some of the lines that I mean:
````
PHP Notice:  Uninitialized string offset: -9 in a/some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 309
PHP Notice:  Uninitialized string offset: -12 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -11 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -10 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -9 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 309
PHP Notice:  Uninitialized string offset: -12 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -11 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -10 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 316
PHP Notice:  Uninitialized string offset: -9 in /some/path/bin/inc/third-party/PHPExcel/Shared/OLERead.php on line 309
````
If PHPExcel sees some of those, it should treat is ad corrupted/damged input file and skip further loading attempts as with a valid file no such E_NOTICE is being shown.

Back to my previous comment, if you hide E_NOTICE (e.g. via php.ini) you won't see this bug ...
